### PR TITLE
Add a `internalConfiguration` admin restful endpoint for query internal configuration

### PR DESCRIPTION
**Pull request by sijie**
### Motivation

Expose `backend` information through admin restful endpoint. so other components can use it to get the backend info.

### Modifications

add `backend` in admin

### Result

application can use `backend` admin endpoint to get backend information.

### CODE REVIEW

**Comment by yush1ga** [comment]
I think the class name `Backend` and `BackendData` is ambiguous.
As these handle only zookeeper's data, these should be named `zkBackendData` or something.

**Comment by rdhabalia** [comment]
> I think the class name Backend and BackendData are ambiguous.

or, we can use `InternalConfiguration/InternalData` instead of Backend because for topic's internal stats we use `internal-stats` term so, internal may give more meaning?

**Comment by sijie** [comment]
@yush1ga @rdhabalia the combination of zkservers and ledgersrootpath is actually the "bookkeeper service url", zkservers indicates where pulsar stores its metadata and globalZooKeeperServers indciates where pulsar stores policies data. so all these 3 settings together are actually describe the "backend" for pulsar brokers.

the purpose of this PR is to provide a method for application to know which zk, bk that a pulsar cluster is using. I am fine with changing the name to a proper name that you think it is better.

so which name I should be using?

InternalConfiguration/InternalData ?
Or? 

If we can get a consensus about the naming, I can make the modification.

/cc @merlimat

**Comment by rdhabalia** [comment]
> the purpose of this PR is to provide a method for application to know which zk, bk that a pulsar cluster is using.
InternalConfiguration/InternalData ?

I think we can add more configurations other than zk in future as it can surface internal configuration. So, should can keep it a generic  name eg: `InternalConfiguration`?

**Comment by sijie** [comment]
Okay will change it to InternalConfiguration

**Comment by sijie** [comment]
@rdhabalia @yush1ga I renamed "backend" to "internalConfiguration". Can you guys review this change again?

**Comment by sijie** [comment]
@rdhabalia @yush1ga can any of you review this again?

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java [hunk 1]
```diff
@@ -154,6 +156,7 @@ public PulsarAdmin(URL serviceUrl, ClientConfiguration pulsarConfig) throws Puls
         WebTarget root = client.target(serviceUrl.toString());
         web = root.path("/admin");
 
+        this.internalConfiguration = new InternalConfigurationImpl(web, auth);
```
Comment (rdhabalia) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: should it be part of `brokers` as this configuration has been configured at broker (eg: `pulsar().getConfiguration().getxxx()`?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@Path("/internal-configuration")
+@Api(value = "/internal-configuration", description = "InternalConfiguration admin apis", tags = "internal-configuration")
+@Produces(MediaType.APPLICATION_JSON)
+public class InternalConfiguration extends AdminResource {
+
+    @GET
+    @ApiOperation(value = "Get the internal configuration data", response = InternalConfigurationData.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 403, message = "Don't have admin permission")
```
Comment (rdhabalia) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java]: in which scenario it returns 403?

**Comment by rdhabalia** [comment]
LGTM.. just few minor comments..

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java (hunk 2 above)
Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java]: isn't accessing admin rest endpoint requires authentication? 403 is returned when client is not authenticated to access admin rest endpoint, no?

please correct me if I am wrong.

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java (hunk 1 above)
Comment (sijie) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: - I was following how brokerStats is organized, which is not part of broker as well. 
- "brokers" is more about "brokers" in a cluster, not about internal stuffs for a broker. configuration here is more about "confiugration" in either cluster/broker level. I am feeling it is much better to have configuration stuffs in a separate endpoint.

let me know what is your opinons

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java (hunk 2 above)
Comment (rdhabalia) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java]: yes, that's correct. I meant `code = 403, message = "Don't have admin permission"` if we think zk-endpoint information should not expose to user and should be accessed by only admin then we can  add validation `AdminResource::validateSuperUserAccess()`

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java (hunk 1 above)
Comment (rdhabalia) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: Ok..sure.. I am not strongly against of having separate resource.
Actually brokers already has `DynamicConfiguration` endpoint as well and internal-config also looks like config which is configured in broker rather a config/policies persisted in zk. So, felt to be part of brokers.

**Comment by sijie** [comment]
@rdhabalia I addressed your comments

1) get rid of 403
2) move `internal-configuration` to be part of `brokers`

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
retest this please

**Pull request merged**